tests/subscriptions/test_scheduler_processing_strategy.py

```diff
@@ -418,3 +438,130 @@ def test_tick_buffer_with_commit_strategy() -> None:
+    )
+
+    future: Future[Message[KafkaPayload]] = Future()
+
+    queue.append(tick_message, deque([future]))
+
+    assert len(queue) == 1
+    assert queue.peek() == TickSubscription(tick_message, future, should_commit=True)
+    assert queue.popleft() == TickSubscription(tick_message, future, should_commit=True)
+    assert len(queue) == 0
+
+
+def test_produce_scheduled_subscription_message() -> None:
+    epoch = datetime(1970, 1, 1)
+    metrics_backend = TestingMetricsBackend()
+    partition_index = 0
+    entity_key = EntityKey.EVENTS
+    topic = Topic("scheduled-subscriptions-events")
+    partition = Partition(topic, partition_index)
+
+    clock = TestingClock()
+    broker_storage: MemoryMessageStorage[KafkaPayload] = MemoryMessageStorage()
+    broker: Broker[KafkaPayload] = Broker(broker_storage, clock)
+
+    broker.create_topic(topic, partitions=1)
+
+    producer = broker.get_producer()
+    consumer = broker.get_consumer("group")
+    consumer.subscribe([topic])
+
+    store = RedisSubscriptionDataStore(
+        redis_client, entity_key, PartitionId(partition_index)
+    )
+    store.create(
+        uuid.uuid4(),
+        SnQLSubscriptionData(
+            project_id=1,
+            time_window=timedelta(minutes=1),
+            resolution=timedelta(minutes=1),
+            query="MATCH events SELECT count()",
+            entity_subscription=EventsSubscription(data_dict={}),
+        ),
+    )
+
+    schedulers = {
+        partition_index: SubscriptionScheduler(
+            store,
+            PartitionId(partition_index),
+            cache_ttl=timedelta(seconds=300),
+            metrics=metrics_backend,
+        )
+    }
+
+    commit = mock.Mock()
+
+    strategy = ProduceScheduledSubscriptionMessage(
+        entity_key,
+        schedulers,
+        producer,
+        KafkaTopicSpec(SnubaTopic.SUBSCRIPTION_SCHEDULED_EVENTS),
+        commit,
+    )
+
+    message = Message(
+        partition,
+        1,
+        CommittableTick(
+            Tick(
+                0,
+                offsets=Interval(1, 3),
+                timestamps=Interval(epoch, epoch + timedelta(minutes=2)),
+            ),
+            True,
+        ),
+        epoch,
+        2,
+    )
+
+    strategy.submit(message)
+
+    # 2 subscriptions should be scheduled (1 for each minute)
+
+    codec = SubscriptionScheduledTaskEncoder(entity_key)
+
+    # First message at epoch
+    first_message = broker_storage.consume(partition, 0)
+    assert first_message is not None
+    assert codec.decode(first_message.payload).timestamp == epoch
+
+    # Second message at epoch + 1 minute
+    second_message = broker_storage.consume(partition, 1)
+    assert second_message is not None
+    assert codec.decode(second_message.payload).timestamp == epoch + timedelta(
+        minutes=1
+    )
+
+    # No 3rd message
+    assert broker_storage.consume(partition, 2) is None
+
+    # Close the strategy
+    strategy.close()
+    strategy.join()
```

Comment:
SO this is a good integration test, but it is missing a few cases to test:
- verify commit happens
- test two subscriptions to ensure the right one is produced.

I think a better way to do this is test multiple use cases with the in memory storage (multiple subscriptions, multiple ticks, multiple partitions, etc.)
Then run one single test on the real Kafka so you can see if the future work properly.